projects/evm-compatible-bridge/design.md

```diff
@@ -0,0 +1,373 @@
+
+The EVM provides events (or logs) which can be used to communicate a new proposal and request for signatures, but for the MultiSig wallet on Hathor the communication needs to be handled in a different way.
+To achieve this we introduce a coordinator service, which will listen for transactions in the MultiSig wallet and coordinate which utxos will be spent for each transaction.
+This coordinator service will also provide some APIs so admins can issue commands to the federation and so admins can check the state of the bridge.
+
+## Hathor MultiSig
+
+Hathor MultiSig standard relies on a pay-to-script-hash (P2SH) model, this means that for any data to be written on-chain (e.g. transactions, token create/mint/melt operations) the signatures for this "write" must already be collected.
+
+To handle operations on the MultiSig wallet each participant will run a copy of a headless wallet initialized with its seed and MultiSig configuration.
+
+We will have a single instance of a coordinator service that all members of the federation will interact with, this coordinator service will be a certerpiece to avoid conflicts in utxo choosing which can block operations if not handled carefully.
+
+Each member of the federation will run a headless wallet configured with the MultiSig configuration and his seed, while the coordinator service will run a headless with the same MultiSig configuration but without a seed since it is not a member of the federation.
+
+For the coordinator service to be notified that a transaction was sent to the "federation" MultiSig wallet we will use the queue plugin of the headless which enables real time notification of a received transaction (the queue can be SQS or rabbitMQ depending on where the service will be deployed).
+
+The federation will have to inspect any transaction proposed by the coordinator service and check if it is valid, to achieve this we need an API on the headless to that can return the transaction information with metadata (balance to our wallet, decoded data outputs, etc.).
+
+## Bridge crossing fees
+
+The fee percentage will be configured on the EVM contract and will be applied to all tokens crossing the bridge, the fee tokens will be kept on the chain their native of.
+
+For clarification, we will list the operations and where the fees are kept:
+
+- Hathor -> EVM (Hathor native)
+  - The fee is deducted from the amount of tokens being crossed and kept on the MultiSig wallet in Hathor.
+  - The admin can request the withdrawal from the wallet by using the coordinator service, which will check that we are not withdrawing more than the collected fees.
+  - For EVM native tokens, we will only melt the amount of tokens minus fees, so the fees will automatically be kept on the MultiSig wallet.
```

Comment:
I did not understand this. I thought we were talking about Hathor native tokens here.